src/escrows/DolaFraxPyUSDEscrow.sol

```diff
@@ -0,0 +1,239 @@
+            revert OnlyBeneficiaryOrAllowlist();
+        _;
+    }
+
+    event AllowClaim(address indexed allowedAddress, bool allowed);
+
+    /**
+    @notice Initialize escrow with a token
+    @dev Must be called right after proxy is created.
+    @param _token The IERC20 token representing the governance token
+    @param _beneficiary The beneficiary who cvxCRV is staked on behalf
+    */
+    function initialize(IERC20 _token, address _beneficiary) public {
+        if (market != address(0)) revert AlreadyInitialized();
+        market = msg.sender;
+        token = _token;
+        token.approve(address(booster), type(uint).max);
+        token.approve(address(yearn), type(uint).max);
+        beneficiary = _beneficiary;
+    }
+
+    /**
+    @notice Withdraws the wrapped token from the reward pool and transfers the associated ERC20 token to a recipient.
+    @dev Will first try to pay from the escrow balance, if not enough or any, will try to pay the missing amount from convex, then from yearn if needed
+    @param recipient The address to receive payment from the escrow
+    @param amount The amount of ERC20 token to be transferred.
+    */
+    function pay(address recipient, uint amount) public {
+        if (msg.sender != market) revert OnlyMarket();
+        uint256 tokenBal = token.balanceOf(address(this));
+        uint256 missingAmount = amount > tokenBal ? amount - tokenBal : 0;
+        // If there are enough tokens in the escrow, transfer the amount
+        if (tokenBal >= amount) {
+            token.safeTransfer(recipient, amount);
+        } else if (stakedBalance >= missingAmount) {
+            // If there are enough staked tokens in convex, withdraw the amount from convex
+            stakedBalance -= missingAmount;
+            rewardPool.withdrawAndUnwrap(missingAmount, false);
+            token.safeTransfer(recipient, amount);
+        } else if (
+            YearnVaultV2Helper.collateralToAsset(
+                yearn,
+                yearn.balanceOf(address(this))
+            ) >= missingAmount
+        ) {
+            // If there are enough tokens in Yearn, withdraw the amount from Yearn
+            uint256 withdrawAmount = YearnVaultV2Helper.assetToCollateral(
+                yearn,
+                missingAmount
+            );
+            // Withdraw from Yearn
+            uint256 assetAmount = yearn.withdraw(withdrawAmount, address(this));
+            // Transfer the amount to the recipient
+            token.safeTransfer(recipient, assetAmount + tokenBal);
+        }
+    }
+
+    /**
+    @notice Get the token balance of the escrow
+    @return Uint representing the staked balance of the escrow
+    */
+    function balance() public view returns (uint) {
+        return
+            stakedBalance +
+            YearnVaultV2Helper.collateralToAsset(
+                yearn,
+                yearn.balanceOf(address(this))
+            ) +
+            token.balanceOf(address(this));
+    }
```

Comment:
This balance function can cause problems, as the pay function currently can only transfer `max(stakedBalance + tokenBal, yearnBalance + tokenBal)`

Reply:
In the worst case scenario it could be used to trigger a `pay(...)` call that results in no tokens being transferred, which in turn could potentially be used to erase your own debt through self-liquidations without losing assets.